packages-ts/gauntlet-terra-contracts/src/commands/abstract/batchWrapper.ts

```diff
@@ -0,0 +1,62 @@
+import AbstractCommand, { makeAbstractCommand } from '.'
+import { Result, WriteCommand } from '@chainlink/gauntlet-core'
+import { TerraCommand, TransactionResponse, logger } from '@chainlink/gauntlet-terra'
+import { AccAddress, LCDClient, Msg } from '@terra-money/terra.js'
+import { prompt } from '@chainlink/gauntlet-core/dist/utils'
+import { ExecutionContext } from './executionWrapper'
+import { parseJSON } from '@chainlink/gauntlet-terra/dist/lib/rdd'
+
+export const wrapCommand = (command) => {
+  return class BatchCommand extends TerraCommand {
+    static id = `${command.id}:batch`
+    subCommands: any[]
+
+    constructor(flags, args) {
+      super(flags, args)
+    }
+
+    buildCommand = async (flags, args): Promise<TerraCommand> => {
+      const input = flags.input ? flags.input : parseJSON(flags.inputFile, 'BatchInputFile')
+
+      this.subCommands = await Promise.all(
+        input.map(async (individualInput) => {
+          const newFlags = { ...flags, input: individualInput }
+
+          let c = new command(newFlags, args) as TerraCommand
+          await c.invokeMiddlewares(c, c.middlewares)
+          c = c.buildCommand ? await c.buildCommand(newFlags, args) : c
+          return c
+        }),
+      )
+      return this
+    }
+
+    makeRawTransaction = async (signer: AccAddress) => {
+      return await Promise.all(this.subCommands.map(async (element) => (await element.makeRawTransaction(signer))[0]))
```

Comment:
We should get every message from the underlying commands, not only the first one. Probably this is the way:
```suggestion
      const rawTxs = (await Promise.all(this.subCommands.map((c) => 
      c.makeRawTransaction(signer)))).reduce((agg, txs) => [
        ...agg,
        ...txs,
      ])
      return rawTxs
```

Reply:
Oh, I did this because I redefined `makeRawTransaction` to return an array of messages, but the array only has one element so I just extract that. But either way, yours is cleaner :)

Reply:
We should not expect to receive only one message, even if every command is doing that now, that could change